hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/volume/OMVolumeRequest.java

```diff
@@ -106,6 +106,13 @@ protected UserVolumeInfo addVolumeToOwnerList(UserVolumeInfo volumeList,
       objectID = volumeList.getObjectID();
     }
 
+    // Sanity check, a user should not own same volume twice
+    //  TODO: May want to remove this due to perf if user owns a lot of volumes.
+    if (prevVolList.contains(volume)) {
+      throw new IOException("Invalid operation: User " + owner +
+          " is about to own a same volume " + volume + " twice!" +
+          " Check for DB consistency error.");
+    }
 
     // Add the new volume to the list
     prevVolList.add(volume);
```

Comment:
Instead of throwing exception here, I was wondering if we can instead perform the subsequent "add new volume to list" if !prevVolList.contains(volume) and complement with WARN log.

```suggestion
    // Avoid adding a user to the same volume twice
    if (!prevVolList.contains(volume)) {
      // Add the new volume to the list
      prevVolList.add(volume);
      UserVolumeInfo newVolList = UserVolumeInfo.newBuilder()
          .setObjectID(objectID)
          .setUpdateID(txID)
          .addAllVolumeNames(prevVolList).build();
    }
```

Reply:
Will do. Thanks :)